docs/book/how-to/popular-integrations/azure-guide.md

```diff
@@ -0,0 +1,196 @@
+
+![Azure App Registrations](../../.gitbook/assets/azure_1.png)
+
+![Azure App Registrations](../../.gitbook/assets/azure_2.png)
+
+Once you create the service principal, you will get an Application ID and 
+Tenant ID as they will be needed later.
+
+Next, go to your service principal and click on the `Certificates & secrets` in
+the `Manage` menu. Here, you have to create a client secret. Note down the 
+secret value as it will be needed later.
+
+![Azure App Registrations](../../.gitbook/assets/azure_3.png)
+
+## 2. Create a service connector
+
+Once you have created the service principal and the client secret, you can 
+go ahead and create [a ZenML Azure Service Connector](../../how-to/auth-management/azure-service-connector.md).
+
+```bash
+zenml service-connector register azure_connector --type azure \
+  --auth-method service-principal \
+  --client_secret=<CLIENT_SECRET> \
+  --tenant_id=<TENANT_ID> \
+  --client_id=<APPLICATION_ID>
+```
+
+You will use this service connector later on to connect your components with 
+proper authentication.
+
+## 3. Create a resource group and the AzureML instance
+
+Now, you have to [create a resource group on Azure](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal).
+To do this, go to the Azure portal and go to the `Resource Groups` page and 
+click `+ Create`. 
+
+![Azure Resource Groups](../../.gitbook/assets/azure_4.png)
+
+Once the resource group is created, go to the overview page of your new resource
+group and click `+ Create`. This will open up the marketplace where you can 
+select a variety of resource to create. Look for `Azure Machine Learning`.
+
+![Azure Marketplace](../../.gitbook/assets/azure_5.png)
+
+Select it, and you will start the process of creating an AzureML workspace. 
+As you can see from the `Workspace details`, AzureML workspaces come equipped 
+with a storage account, key vault, and application insights. It is highly 
+recommended that you create a container registry as well.
+
+![AzureML Workspace Details](../../.gitbook/assets/azure_6.png)
+
+## 4. Create Stack Components
+
+In order to run any workflows on Azure using ZenML, you need an artifact store,
+an orchestrator and a container registry.
+
+### Artifact Store (Azure Blob Storage)
+
+For the artifact store, we will be using the storage account attached to our
+AzureML workspace. But before registering the component itself, you have to 
+create a container for blob storage. To do this, go to the corresponding 
+storage account in your workspace and create a new container:
+
+![Azure Blob Storage](../../.gitbook/assets/azure_7.png)
+
+Once you create the container, you can go ahead, register your artifact 
+store using its path and connect it to your service connector:
+
+```bash 
+zenml artifact-store register azure_artifact_store -f azure  \
+  --path=<PATH_TO_YOUR_CONTAINER>
+  
+zenml artifact-store connect azure_artifact_store -c azure_connector
```

Comment:
Our `zenml <component> register` commands have a `--connector` argument which avoids having this as a separate call

Reply:
Same for the other components below

Reply:
You are right. Changed the instructions here.